persistence/src/main/scala/za/co/absa/spline/persistence/model/operations.scala

```diff
@@ -17,43 +17,52 @@
 package za.co.absa.spline.persistence.model
 
 sealed trait Operation extends Vertex {
+  def name: Option[Operation.Name]
   def params: Map[String, Any]
   def extra: Map[String, Any]
-  def `type`: String
+  def `type`: Operation.Type
```

Comment:
yes, a lot of confusion here... Originally the logic was to have all technical (non-business) props to be prefixed with underscore, just to avoid confusion in situations like when you add a business property with a similar name. For example operation type "Filter" vs type "transformation" or vs type "za.co.absa....operation.FilterOperation" -- here every type means different things and is supposed to be analyzed on different application layers. It worked well before and in general, but in case with Arango, since we don't have a control on how JSON fields are mapped on the class properties in Arango driver this convention leaked into the domain model and started making noise there. And also `_` is a reserved for Arango stuff, so I think we need to get rid of using that prefix convention.